test/test_tleap.py

```diff
@@ -0,0 +1,246 @@
+    waters = np.random.randint(1000, 10000)
+    log.debug('Trying {} waters in an isometric box...'.format(waters))
+    sys = System()
+    sys.template_file = 'cb6-but/tleap_solvate.in'
+    sys.output_path = 'cb6-but'
+    sys.loadpdb_file = 'cb6-but.pdb'
+    sys.buffer_target = waters
+    sys.output_prefix = 'solvate'
+    sys.pbc_type = 'cubic'
+    sys.build()
+    grepped_waters = sp.check_output(["grep -oh 'WAT' ./cb6-but/solvate.prmtop | wc -w"], shell=True)
+    assert int(grepped_waters) == waters
+    rm_solvated_files()
+
+@pytest.mark.slow
+def test_solvation_spatial_size():
+    """ Test that we can solvate CB6-BUT with an buffer size in Angstroms. """
+    # this test is a sorta dumb
+    random_int = np.random.randint(10, 20)
+    random_size = random_int * np.random.random_sample(1) + random_int
+    log.debug('Trying buffer size of {} A...'.format(random_size[0]))
+    sys = System()
+    sys.template_file = 'cb6-but/tleap_solvate.in'
+    sys.output_path = 'cb6-but'
+    sys.loadpdb_file = 'cb6-but.pdb'
+    sys.buffer_target = '{0:1.4f}A'.format(random_size[0])
+    sys.output_prefix = 'solvate'
+    sys.pbc_type = 'cubic'
+    sys.build()
+    grepped_waters = sp.check_output(["grep -oh 'WAT' ./cb6-but/solvate.prmtop | wc -w"], shell=True)
+    assert int(grepped_waters) == sys.target_waters
+    rm_solvated_files()
+
+@pytest.mark.slow
+def test_solvation_potassium_control():
+    """ Test there is no potassium by default. A negative control. """
+    waters = np.random.randint(1000, 10000)
+    log.debug('Trying {} waters with potassium...'.format(waters))
+    sys = System()
+    sys.template_file = 'cb6-but/tleap_solvate.in'
+    sys.output_path = 'cb6-but'
+    sys.loadpdb_file = 'cb6-but.pdb'
+    sys.buffer_target = waters
+    sys.output_prefix = 'solvate'
+    sys.counter_cation = 'K+'
+    sys.build()
+    potassium = sp.check_output(["grep -oh 'K+' ./cb6-but/solvate.prmtop | wc -w"], shell=True)
+    assert int(potassium) == 0
+    rm_solvated_files()
+
+@pytest.mark.slow
+def test_solvation_with_additional_ions():
+    """ Test that we can solvate CB6-BUT with additional ions. """
+    waters = np.random.randint(1000, 10000)
+    cations = ['LI', 'Na+', 'K+', 'RB', 'CS']
+    anions = ['F', 'Cl-', 'BR', 'IOD']
+    n_cations = np.random.randint(1, 10)
+    n_anions = np.random.randint(1, 10)
+    random_cation = random.choice(cations)
+    random_anion = random.choice(anions)
+    log.debug('Trying {} waters with additional ions...'.format(waters))
+    sys = System()
+    sys.template_file = 'cb6-but/tleap_solvate.in'
+    sys.output_path = 'cb6-but'
+    sys.loadpdb_file = 'cb6-but.pdb'
+    sys.buffer_target = waters
+    sys.output_prefix = 'solvate'
+    sys.neutralize = False
+    sys.add_ions = [random_cation, n_cations, random_anion, n_anions]
+    sys.build()
+    # These should come in the RESIDUE_LABEL region of the prmtop and be before all the water.
```

Comment:
Nice, so this allows us to get rid of that `# Have to think about what to do here...` block below?

Reply:
I was confused about this to begin with ... not sure why you `#Have to think about what to do here ...`

Reply:
Fair to delete it.

Reply:
Done